GitHub_Learning/Feature/Dashboard/Resource/CryptocurrencyPriceResource.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  cryptoPriceResource.swift
+//  GitHub_Learning
+//
+//  Created by Amin Ashhab on 20.12.2020.
+//  Copyright © 2020 Amin_Second_Test_Project. All rights reserved.
+//
+
+import Foundation
+import CleanCore
+
+struct CryptoPriceRequest {}
+
+typealias CryptoPriceResponse = Cryptocurrency  
+
+protocol CryptocurrencyPriceResource {
+    func getCryptocurrencyPrice() throws -> Cryptocurrency
+}
+
+class CryptocurrencyPriceResourceImpl: BaseHttpRemoteResource<CryptoPriceRequest, CryptoPriceResponse>, CryptocurrencyPriceResource {
+    private let baseUrlConfig: BaseUrlConfig
+    private let cryptocurrencyPriceParser: CryptocurrencyParser
+    
+    init(
+        networkClient: HttpNetworkClient,
+        errorParser: BusinessErrorParser,
+        creationStep: RequestCreationStep,
+        deserializationStrategy: ResponseBodyDeserializationStrategy,
+        baseUrlConfig: BaseUrlConfig,
+        cryptocurrencyPriceParser: CryptocurrencyParser
+    ) {
+        self.baseUrlConfig = baseUrlConfig
+        self.cryptocurrencyPriceParser = cryptocurrencyPriceParser
+        super.init(
+            networkClient: networkClient,
+            errorParser: errorParser,
+            creationStep: creationStep,
+            deserializationStrategy: deserializationStrategy
+        )
+    }
+    
+    func getCryptocurrencyPrice() throws -> Cryptocurrency {
+        let request = CryptoPriceRequest()
+        return try makeRequest(request)
+    }
+    
+    override func resourcePath(_ request: CryptoPriceRequest) throws -> String {
+        return baseUrlConfig.baseUrl() + "api.coingecko.com/api/v3/coins/markets?&order=market_cap_desc&per_page=100&page=1&sparkline=false"
```

Comment:
Ten base url tady nepotrebujes. Ten bys pouzil, pokud bys mel server, na ktery mas vic nez jedno API a chteli jsi sdilet prvni cast URL. tj `api.coingecko.com/api/v3` v tomto pripade.

Reply:
Já tady moc nevim právě, hledal jsem to předtim a dostal jsem se k tomu že jen to natvrdo naijectovaný a vyhazuje to jen https://, a v cisu to používali ale nenašel jsem že by to nějak preinjectovávali aby to vyhozovalo něco jinýho.  Asi taky možná by bylo lepší pak se na to společně kouknout jestli si rozumíme. ale klidně tam to https: můžu napsat ručně.

Reply:
Asi je to zbytečný no máš pravdu.